src/crawlee/_types.py

```diff
@@ -46,6 +46,20 @@
 
 RequestTransformAction: TypeAlias = Literal['skip', 'unchanged']
 
+EnqueueStrategy: TypeAlias = Literal['all', 'same_domain', 'same_hostname', 'same_origin']
+"""Specify the strategy for determining which links to extract and enqueue.
+
+Strategies:
```

Comment:
I discussed this with @barjin and it seems that we won't be able to display this docs anywhere in the API reference, which is a disadvantage of enums (those get their own generated pages). It's probably still worth it, I just want to make sure we make an informed decision.

Reply:
Ok, we can explain it in the enqueue links function docstring then.

Reply:
That works for me. The only drawback I see is that if we added a new `EnqueueStrategy` (unlikely), we would have to document it in the function docstring too, which is easy to miss.

Reply:
updated